sacrebleu/tokenizers/tokenizer_spm.py

```diff
@@ -0,0 +1,32 @@
+# -*- coding: utf-8 -*-
+
+import os
+import urllib.request
+
+from .tokenizer_none import NoneTokenizer
+
+class TokenizerSPM(NoneTokenizer):
+    def signature(self):
+        return 'spm'
+
+    def __init__(self):
+        try:
+            import sentencepiece as spm
+        except (ImportError, ModuleNotFoundError):
+            raise ImportError(
+                '\n\nPlease install the sentencepiece library for SPM tokenization:'
+                '\n\n  pip install sentencepiece '
+            )
+        self.sp = spm.SentencePieceProcessor()
+        if not os.path.exists('sacrebleu_tokenizer_spm.model'):
+            url = "https://dl.fbaipublicfiles.com/fairseq/models/flores/sacrebleu_tokenizer_spm.model"
+            urllib.request.urlretrieve(url, 'sacrebleu_tokenizer_spm.model')
+        self.sp.Load("sacrebleu_tokenizer_spm.model")
+
```

Comment:
A few comments here:

- This should be saved to `~/.sacrebleu/models/`, not the current directory
- Can you add a checksum test and locking? An early sacrebleu bug was people running multiple at the same time, and it would download the test sets, overwriting each other. I think you could refactor `download_test_set` to have a `download_file(uri, local_path, checksum)`, and then this could use that function, too
- The model file name should be more description, ideally containing the checksum, I think?
- It would be helpful to log a message that the download is happening, since it's likely to take some time

Reply:
Hi @ngoyal2707, just pinging you here in case you missed these comments.